### DISCUSSION TITLE
CARLA's event camera vs rpg_esim

### DISCUSSION THREAD

**Discussion by qhaas** [carla-simulator/carla#4634] [Q&A]
I was in need of an event (i.e. DVS / Neuromorphic) camera simulator.  I read [this paper](http://rpg.ifi.uzh.ch/docs/CORL18_Rebecq.pdf) from the RPG lab at University of Zurich and I started fiddling with their [open-source (UnrealEngine based) event camera simulator](https://github.com/uzh-rpg/rpg_esim).  The paper makes a good argument as to the performance of their event camera simulator (e.g. adaptive sampling).

I have been a CARLA user for years, so I was excited to see an event camera added with [CARLA 0.9.10](https://carla.org/2020/09/25/release-0.9.10/):
```
Added Dynamic Vision Sensor (DVS) camera based on ESIM simulation http://rpg.ifi.uzh.ch/esim.html
```
Later, it would appear that the RPG lab is contributing to CARLA with [CARLA 0.9.12](https://carla.org/2021/08/02/release-0.9.12/):
```
For the Optical Flow Camera, we thank the team at RPG Lab at the University of Zurich!
```

CARLA is a more general simulator than rpg_esim and easier to setup (especially on distributions where getting ROS working is tricky).  Given RPG is apparently contributing to CARLA, has the Event Camera in CARLA reached parity with rpg_esim?

**Comment by edelaney11** [comment]
Did you ever find an answer for this?

**Reply by qhaas** [reply]
No, we have not done a formal validation study comparing the realism of carla's dvs implementation to rpg_esim, nor to those of real event cameras.

Given event (aka dvs aka neuromorphic) cameras are currently broken in the latest release ( see issue #8724 ), said sensor likely isn't a high priority upstream.  We are currently operating under the assumption that a higher FPS means less interpolation for the 'intermediate events' between frames (assuming CARLA diffs frames to create a baseline for events) and thus increased realism ( see issue #6549 ).

**Reply by edelaney11** [reply]
That makes sense, thanks for your help!
